File: Android_Zone_Test/src/com/example/mylib_test/activity/system/SystemMainActivity.java
package com.example.mylib_test.activity.system;

import com.example.mylib_test.R;

import and.utils.activity_fragment_ui.ToastUtils;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.View.OnClickListener;

public class SystemMainActivity extends Activity implements OnClickListener{
	public static int ResponseCode=11111;
	public static int RequestCode=22222;
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.a_system_test);
	}
	
	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		getMenuInflater().inflate(R.menu.main, menu);
		return true;//true创建的菜单才能显示
	}
	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		switch (item.getItemId()) {
		case R.id.action_settings:
			
			break;
		case R.id.sb:
			
			break;

		default:
			break;
		}
		ToastUtils.showLong(this, "干啥啊别碰我");
		return super.onOptionsItemSelected(item);
	}
	@Override
	public void onClick(View v) {
		switch (v.getId()) {
		case R.id.bt_activityResult:
			startActivityForResult(new Intent(this,ResultActivity2.class), SystemMainActivity.RequestCode);
			break;
		case R.id.bt_baseResult:
			startActivityForResult(new Intent(this,ResultBase1.class), SystemMainActivity.RequestCode);
			break;
		case R.id.bt_webView:
			startActivity(new Intent(this,WebViewAcitivity.class));
			break;
		case R.id.crash:
			throw new NullPointerException("crash test");
		default:
			break;
		}
	}
	@Override
	protected void onActivityResult(int requestCode, int resultCode, Intent data) {
		super.onActivityResult(requestCode, resultCode, data);
		System.out.println("first onActivityResult____ requestCode:"+requestCode+"\t resultCode"+resultCode);
	}

}

File: Android_Zone_Test/src/com/example/mylib_test/activity/system/SystemMainActivity.kt
package com.example.mylib_test.activity.system

import com.example.mylib_test.R

import com.zone.lib.utils.activity_fragment_ui.ToastUtils

import android.content.Intent
import android.view.Menu
import android.view.MenuItem
import android.view.View
import com.example.mylib_test.activity.animal.PixelsAcitivity
import com.example.mylib_test.activity.statu.StatuMainActivity
import com.example.mylib_test.activity.statu.StatueBarModeActivity
import com.zone.lib.base.controller.activity.BaseFeatureActivity
import com.zone.lib.base.controller.activity.controller.ActionBarActivityController
import com.zone.lib.base.controller.activity.controller.ShowState

class SystemMainActivity : BaseFeatureActivity() {

    companion object {
        var ResponseCode = 11111
        var RequestCode = 22222
    }

    override fun initDefaultConifg(){
        getController(ActionBarActivityController::class.java)?.initFirst(ShowState.ShowTitle)
    }

    override fun setContentView() {
        setContentView(R.layout.a_system_test)
    }

    override fun initData() {
    }

    override fun setListener() {
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main, menu)
        return true//true创建的菜单才能显示
    }


    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item?.itemId) {
            R.id.action_settings -> {
            }
            R.id.sb -> { }
            else -> {
            }
        }
        ToastUtils.showLong(this, "干啥啊别碰我")
        return super.onOptionsItemSelected(item)
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.bt_activityResult -> startActivityForResult(Intent(this, ResultActivity2::class.java), SystemMainActivity.RequestCode)
            R.id.bt_layoutInflater -> startActivityForResult(Intent(this, LayoutInflaterActivity::class.java), SystemMainActivity.RequestCode)
            R.id.crash -> throw NullPointerException("crash test")
            R.id.bt_system_status ->  startActivity(Intent(this, StatuMainActivity::class.java))
            R.id.bt_statuBarMode ->  startActivity(Intent(this, StatueBarModeActivity::class.java))
            else -> {
            }
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        println("first onActivityResult____ requestCode:$requestCode\t resultCode$resultCode")
    }

}
